### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by srueg**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the ./CHANGELOG.md.

### CODE REVIEW

**Comment by srueg** [comment]
Bug report to investigate the issues: #203

**Comment by simu** [comment]
PR to gracefully handle inventory_path change in Kapitan 0.29.3: #212

**Comment by srueg** [comment]
Now with https://github.com/deepmind/kapitan/releases/tag/v0.29.4 we can finally proceed with this.

File: tools/Dockerfile.kapitan [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 FROM docker.io/alpine:3.12
 
-ENV KAPITAN_VERSION=v0.29.1
+ENV KAPITAN_VERSION=v0.29.2
```
Comment (simu) [inline tools/Dockerfile.kapitan:5]: This probably should be bumped to `v0.29.4` also

**Comment by srueg** [comment]
I get the following error regarding `inventory_path`:
```
Jsonnet error: failed to compile /Users/srueg/Development/VSHN/SYN/commodore/vendor/argocd/component/app.jsonnet:
 RUNTIME ERROR: 'NoneType' object has no attribute 'inventory_path'
	/Users/srueg/Library/Caches/pypoetry/virtualenvs/commodore-rUmS3EGq-py3.9/lib/python3.9/site-packages/kapitan/lib/kapitan.libjsonnet:9:59-100	function <anonymous>
	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:8:13-28	thunk <inv>
	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:9:16-19	thunk <params>
	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:75:16-22	object <anonymous>
	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:(73:13)-(76:4)	object <anonymous>
	/Users/srueg/Development/VSHN/SYN/commodore/vendor/argocd/component/app.jsonnet:37:25-40	object <anonymous>
	During manifestation

Compile error: failed to compile target: argocd
```

**Comment by simu** [comment]
I'm unable to setup a Poetry env on Ubuntu 20.04 (Python 3.8):

```
simon@phoenix:~/work/syn/commodore $ poetry install
Installing dependencies from lock file

[RecursionError]
maximum recursion depth exceeded
simon@phoenix:~/work/syn/commodore $ git branch
* bump-dependencies
```

**Comment by simu** [comment]
> I'm unable to setup a Poetry env on Ubuntu 20.04 (Python 3.8):
> 
> ```
> simon@phoenix:~/work/syn/commodore $ poetry install
> Installing dependencies from lock file
> 
> [RecursionError]
> maximum recursion depth exceeded
> simon@phoenix:~/work/syn/commodore $ git branch
> * bump-dependencies
> ```

Hm, after updating poetry to 1.1.4, install seems to work

**Comment by simu** [comment]
> I get the following error regarding `inventory_path`:
> 
> ```
> Jsonnet error: failed to compile /Users/srueg/Development/VSHN/SYN/commodore/vendor/argocd/component/app.jsonnet:
>  RUNTIME ERROR: 'NoneType' object has no attribute 'inventory_path'
> 	/Users/srueg/Library/Caches/pypoetry/virtualenvs/commodore-rUmS3EGq-py3.9/lib/python3.9/site-packages/kapitan/lib/kapitan.libjsonnet:9:59-100	function <anonymous>
> 	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:8:13-28	thunk <inv>
> 	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:9:16-19	thunk <params>
> 	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:75:16-22	object <anonymous>
> 	/Users/srueg/Development/VSHN/SYN/commodore/vendor/lib/argocd.libjsonnet:(73:13)-(76:4)	object <anonymous>
> 	/Users/srueg/Development/VSHN/SYN/commodore/vendor/argocd/component/app.jsonnet:37:25-40	object <anonymous>
> 	During manifestation
> 
> Compile error: failed to compile target: argocd
> ```

I can't reproduce this error locally, can you provide me the cluster for which you get the error privately?

**Comment by srueg** [comment]
After some testing it seems that this issue only occurs on native macOS installations. Using Docker on macOS works as intended.
Since we don't support native macOS anymore anyway, it's not a blocker for now.

**Comment by simu** [comment]
> After some testing it seems that this issue only occurs on native macOS installations. Using Docker on macOS works as intended.
> Since we don't support native macOS anymore anyway, it's not a blocker for now.

We should probably remove the OS X parts from the "Getting Started" section in the README, if we're officially dropping support for poetry-on-OSX

**Review by simu — APPROVED** [review]

**Pull request merged**